github/resource_github_organization_role_team_assignment.go

```diff
@@ -11,6 +11,8 @@ import (
 
 func resourceGithubOrganizationRoleTeamAssignment() *schema.Resource {
 	return &schema.Resource{
+		DeprecationMessage: "This resource is deprecated in favor of the github_organization_role_team resource.",
```

Comment:
question: Do we have a naming convention for these kinds of resources? I personally think that `github_organization_role_team_assignment` is clearer than `github_organization_role_team`, especially since GitHub uses the "Role assignment" terminology

And a second note: The docs for https://registry.terraform.io/providers/integrations/github/latest/docs/resources/organization_role_team_assignment are more verbose and user-friendly than https://registry.terraform.io/providers/integrations/github/latest/docs/resources/organization_role_team

So, if we want to deprecate `github_organization_role_team_assignment` can we copy over the docs from it? :)